gocourse18/domains/user/repository_test.go

```diff
@@ -0,0 +1,97 @@
+package user
+
+import (
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/stretchr/testify/assert"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+	"testing"
+)
+
+func setupMockDB() (*gorm.DB, sqlmock.Sqlmock, error) {
+	sqlDB, mock, err := sqlmock.New()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	mock.ExpectQuery("SELECT VERSION()").WillReturnRows(sqlmock.NewRows([]string{"VERSION()"}).AddRow("8.0.23"))
+	gormDB, err := gorm.Open(mysql.New(mysql.Config{Conn: sqlDB}), &gorm.Config{})
+	if err != nil {
+		return nil, nil, err
+	}
+	return gormDB, mock, nil
+}
+
+func TestFindOneById(t *testing.T) {
+	db, mock, err := setupMockDB()
+	if err != nil {
+		t.Fatalf("failed to set up mock database: %v", err)
+	}
+	repo := &Repository{db: db}
+
+	expectedUser := User{
+		ID:         1,
+		Name:       "Alice",
+		Surname:    "Johnson",
+		Email:      "alice@example.com",
+		Age:        30,
+		Sex:        "Female",
+		City:       "New York",
+		TaxiCount:  5,
+		Profession: "Engineer",
+	}
+	rows := sqlmock.NewRows([]string{
+		"id", "name", "surname", "email", "age", "sex", "city", "taxi_count", "profession",
+	}).AddRow(
+		expectedUser.ID, expectedUser.Name, expectedUser.Surname, expectedUser.Email,
+		expectedUser.Age, expectedUser.Sex, expectedUser.City, expectedUser.TaxiCount, expectedUser.Profession,
+	)
+
+	mock.ExpectQuery("SELECT \\* FROM `users` WHERE `users`.`id` = \\? ORDER BY `users`.`id` LIMIT \\?").
+		WithArgs(1, 1).
+		WillReturnRows(rows)
+
+	user, err := repo.FindOneById(1)
+	assert.NoError(t, err)
```

Comment:
Для помилок краще використовувати `require` з `github.com/stretchr/testify/require`. Тому що немає сенсу далі перевіряти:
```suggestion
	require.NoError(t, err)
```